### PR TITLE
feat(refresh): implement refresh logic for GitHub OAuth app user credentials

### PR DESCRIPTION
**Pull request by hassan254-prog**
## Describe the problem and your solution
- Currently, we only refresh tokens for GitHub App OAuth, leaving out `userCredentials` that reside in the `connectionConfig`. This PR addresses that by ensuring both are refreshed. Once a connection is created, we determine the earliest `expiresAt` value between the app and user credentials and use that as the effective token expiration time. When Nango refreshes, both the app and user credentials are refreshed accordingly.

## Testing instructions (skip if just adding/editing providers)
1. Create a GitHub App OAuth connection.
2. Observe that the token expiration is based on the earlier of the two `expiresAt` values.
3. After a token refresh occurs, confirm that the expiration is updated again based on the new earliest `expiresAt`.

---

This PR overhauls how GitHub App OAuth connections manage and refresh credentials by ensuring both user and app tokens are properly refreshed and stored. It introduces a new combined credential structure, updates the token expiration logic to use the earliest expiration between the app and user tokens, and comprehensively updates underlying service logic, the OAuth2 client, and related types. Robust error handling and extensive unit tests are added for new scenarios and edge cases.

<details>
<summary><strong>Key Changes</strong></summary>

• Introduced CombinedOauth2AppCredentials type for representing both app and user tokens under Custom (GitHub App OAuth) flows
• Refactored refresh logic to update both app and user tokens; after refresh, credentials are split between connection.credentials (for the app) and connection_config['userCredentials'] (for the user)
• getExpiresAtFromCredentials utility now returns the earliest expiration between app and user tokens for combined credentials; falls back if missing
• Updated getNewCredentials and getFreshOAuth2Credentials logic to accommodate and properly handle combined credentials
• Extended error handling to cover missing user credentials and relevant refresh scenarios
• Updated types and interfaces throughout the codebase to safely carry the new credential structure
• Added comprehensive unit tests for getExpiresAtFromCredentials covering all combinations/edge cases

</details>

<details>
<summary><strong>Affected Areas</strong></summary>

• Connection refresh and storage logic (connection.service.ts, refresh.ts)
• OAuth2 client refresh behavior
• Auth utility logic and type definitions (@nangohq/types, utils.ts)
• Error handling framework
• Unit test coverage for refresh and combined credential logic

</details>

*This summary was automatically generated by @propel-code-bot*

### CODE REVIEW

File: packages/shared/lib/services/connection.service.ts [hunk 1]
```diff
@@ -1003,6 +1005,50 @@ class ConnectionService {
         return Ok(updatedConnection);
     }
 
+    public async refreshAppCredentials(
```
Comment (bodinsamuel) [inline packages/shared/lib/services/connection.service.ts]: can it be named githubApp?

**Review by bodinsamuel — APPROVED** [review]

**Review by bodinsamuel — APPROVED** [review]
